packages/nuxt/src/app/plugins/chunk-reload-immediate.client.ts

```diff
@@ -0,0 +1,16 @@
+import { defineNuxtPlugin } from '../nuxt'
+import { reloadNuxtApp } from '../composables/chunk'
+
+const reloadNuxtApp_ = () => { reloadNuxtApp({ persistState: true }) }
+
+// See https://github.com/nuxt/nuxt/issues/23612 for more context
+export default defineNuxtPlugin({
+  name: 'nuxt:chunk-reload-immediate',
+  setup (nuxtApp) {
+    // Reload when a `chunkError` is thrown
+    nuxtApp.hook('app:chunkError', reloadNuxtApp_)
```

Comment:
If we are reloading while navigating, we should probably reload at the new path rather than the current one.

Reply:
@danielroe ( @BracketJohn )
Can I ask you something? :)
The image of the source would be as follows?

```ts
    nuxtApp.hook('app:chunkError', () => {
      reloadNuxtApp({
        persistState: true,
        path: nuxtApp._route.path
      })
    })
```

Reply:
Yes, much like that! Except, I think, that we might need to track the potential route with a middleware in case the error happens whilst navigating, but before the route has updated.

Reply:
You're absolutely right. We should definitely account for the possibility of an error occurring during navigation before the route is updated. In that case, it might be a good idea to use addRouteMiddleware.
Here is a sample code.

```ts
    const state = useState<{ path: string | null }>('chunk-reload-immediate-path-state', () => ({
      path: null
    }))

    addRouteMiddleware((to) => {
      state.value.path = to.path
    })

    nuxtApp.hook('app:chunkError', () => {
      reloadNuxtApp({
        persistState: true,
        path: state.value.path ?? nuxtApp._route.path
      })
    })
```

I'm just unsure if the middleware's process is executed before `app:chunkError` . 🤔

Reply:
> If we are reloading while navigating, we should probably reload at the new path rather than the current one.

@danielroe good point. I've implemented an approach that remembers the last `currentlyNavigationTo = to.path` using a route middleware (similar to how @noriyuki-shimizu proposed above). If no `currentlyNavigationTo` is known, we fallback to `nuxtApp._route.path`.